giftree-app/giftree
# Fix file structure

A lot of changes were made in this branch:
1. Changed file structure to contain front end files in `giftree/frontend` and backend files in `giftree/backend`. The `package.json` inside `giftree/frontend` is all the frontend and react stuff, vs the `package.json` in the root folder `giftree` is for the backend stuff, therefore you now have to run `npm install` in both locations, more on this is on the updated readme.
2. Made some changes in `server.js` to make this a deployable branch and it works to deploy the frontend and api to heroku.
3. Added email validation in the frontend (this was kind of random but I just wanted to do it before I added this to main).
4. I updated the readme with very thorough instructions on how these changes will affect your development and I will let y'all know what other changes you'll need to make in your respective branches to make sure they don't break.

To test this:
- Check out this branch
`git checkout fix-file-structure`
- Install npm stuff
`npm install`
`cd frontend`
`npm install`
- Create a build
`ionic build`
`cd ..`
- Start the server
`npm start`
- Now go to http://localhost:5000/

Most of these commits are really confusing because I had to keep committing and pushing in order to test the deployment, but the changes overall are pretty minimal.
